torch/_inductor/codegen/wrapper.py

```diff
@@ -1240,6 +1241,26 @@ def _generate(self, is_inference):
 
         return result.getvaluewithlinemap()
 
+    def _finalize_autotune_calls(self):
+        """Delete all autotune-generated tensors after the last use."""
+        to_delete: dict[str, list[str]] = {n: [] for n in self.kernel_autotune_names}
+        for tensor_arg, kernel_name in self.kernel_autotune_example_args.values():
+            to_delete[kernel_name].append(tensor_arg)
+
+        for kernel_name, tensors_to_delete in to_delete.items():
+            if not tensors_to_delete:
+                continue
+
+            for i, line in enumerate(self.kernel_autotune_calls._lines):
+                if isinstance(line, str) and (
+                    m := re.match(rf"( +){kernel_name}.run\(", line)
+                ):
+                    # the m.group(1) call ensures that the indentation is consistent
+                    self.kernel_autotune_calls._lines.insert(
+                        i + 1, f"{m.group(1)}del {', '.join(tensors_to_delete)}"
+                    )
+                    break
```

Comment:
this is strange.. is there a better way of doing this ?

Reply:
@eellison Which part is strange to you? I could definitely eliminate the use of `m.group(1)`; I think the indent is always 4 spaces here. But the other stuff is necessary, I believe.

Reply:
Its strange to me that we have to regex code that we have generated. that information should have been present at one point, we've lost it apparently, and now we are regex-ing it back. I would not expect us to need to regex.

Reply:
I can drop the regex, but I think I'll have to do some form of search over the `kernel_autotune_calls` lines, at least as currently structured. The main bit of info we're missing is about which line number the kernel call happens at, so that we can put the deletion immediately afterward. Let me do one more pass to ensure there's nothing I'm missing that would make that easier, though.

Reply:
I think a cleaner approach is to use something similar to MemoryPlanningLine.

Reply:
@desertfire Good idea! I'll put this back into draft and leave it near the end of the stack until I have time to implement that, so that it won't block the rest of the stack.

Reply:
Done.